src/components/questionsSetScreen/components/chooseMode.js

```diff
@@ -1,23 +1,31 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { Dimensions, StyleSheet, Text, TouchableOpacity, View } from "react-native";
 import { useNavigation } from "@react-navigation/native";
+import { useApi } from "../../../utils/api";
 
 export const ChooseMode = ({
- item,
- currentTopic,
- mainColor,
+  chosenQuestionsSet,
+  mainColor,
   headerBackground,
+                             questionsSets
 }) => {
   const navigation = useNavigation();
+  const api = useApi('http://localhost:1339/');
+  const [questions, setQuestions] = useState([]);
+  useEffect(() => {
+    api.questions(chosenQuestionsSet.name).then((questions) => setQuestions(questions));
```

Comment:
write it like .then((questions) => setQuestions(questions)); -> .then(setQuestions);